components/status_dropdown/index.ts

```diff
@@ -13,18 +13,17 @@ import {get} from 'mattermost-redux/selectors/entities/preferences';
 import {openModal} from 'actions/views/modals';
 import {setStatusDropdown} from 'actions/views/status_dropdown';
 
-import StatusDropdown from 'components/status_dropdown/status_dropdown.jsx';
 import {makeGetCustomStatus, isCustomStatusEnabled, showStatusDropdownPulsatingDot} from 'selectors/views/custom_status';
 import {isStatusDropdownOpen} from 'selectors/views/status_dropdown';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {GlobalState} from 'types/store';
 
-function mapStateToProps(state) {
+import StatusDropdown from './status_dropdown';
+
+function mapStateToProps(state: GlobalState) {
     const currentUser = getCurrentUser(state);
     const getCustomStatus = makeGetCustomStatus();
 
-    if (!currentUser) {
```

Comment:
Can you elaborate on removing this check? If there is no current user -- as unlikely as that may be, it can sometimes trigger during a logout, for example -- will the subsequent checks spuriously fail?

Reply:
Hi @lieut-data , I remove the code to clean the typescript error. 

If I add code checking currentUser:
`if (!currentUser) { 
      return {};
  }
`

the data type of userId, profilePicture, autoResetPref, customStatus, and etc in components/status_dropdown/index.ts will be  **their own data type(string/bool/UserCustomStatus)**  or **undefined**. If the type can be **undefined**, I need to set the  type Props in components/status_dropdown/status_dropdown.tsx to allow undefined.

To resolve this issue, I check another  example of mattermost-webapp typescript code, and there is no logic checking if currentUser exists in file "index.ts". So, I remove the checking in status_dropdown.ts file and test it manually in my local.

But, if I check the javascript code (jsx),  there are several code like this (checking currentUser).
What do you think about this ?  



<img width="1079" alt="Screen Shot 2021-05-03 at 19 59 39" src="https://user-images.githubusercontent.com/28821618/116879073-63fcc780-ac4a-11eb-923c-33ceb9ceb21d.png">

Reply:
I think this is okay, but to be safe I think we should add coalescing operators in front of `currentUser` in this connector so that we never end up in a situation where the app will whitescreen if `currentUser` is `undefined`

Reply:
Hi @devinbinnie  I already add coalescing operator  on several lines